ocaml/lambda/matching.ml

```diff
@@ -2178,6 +2190,7 @@ let get_expr_args_array ~scopes kind head (arg, _mut, _layout) rem =
         (match am with
         | Mutable   -> StrictOpt
         | Immutable -> Alias),
+        Sort.for_predef_param,
```

Comment:
It seemed like overkill to make separate `Sort.foo` definitions for the many predefined types (like `int` and `bool`) that are all values.  So I made one definition, `Sort.for_predef_value` for those.  Similarly, all type parameters of predefined types are currently values, so I made one definition for those `Sort.for_predef_param`.

Perhaps a more descriptive name or a comment would be an adequate compromise?  Or I can go in and make a bunch more aliases for value if you prefer.

Reply:
Ah -- I did not grok the different between `predef_param` and `predef_value`. The problem is that these might change in the future, and they might change at different times. For example, we'll get non-values in arrays much sooner than non-values in lists (which requires overcoming the mixed-block restriction). And array comprehensions will want to be able to extract floats from arrays without allocating (maybe -- might not be worth it). So `Sort.for_int` and `Sort.for_bool` are maybe overkill, but I do think separating out `Sort.for_list_element` from `Sort.for_array_element` is definitely worthwhile. (By the way, I chose `Sort.for_array_get_result` above, not `Sort.for_array_element`, because they might be different! Because of the float array optimization.)

Reply:
OK, that sounds sensible.  I have:
- Eliminated `Sort.for_predef_param`, replacing it with `Sort.for_array_element`, `Sort.for_array_get_result` and `Sort.for_list_element`.
- Put a comment on `for_predef_value` explaining its purpose.

And resolved all the comments about those two things.